stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,598 @@
+// The module map for the Android NDK.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
+  header "SwiftAndroidNDK.h"
+  export *
+}
+
+// The modules for the Bionic C standard library used by
+// the Android NDK.
+module bionic_assert [system] {
```

Comment:
I have the same comment here as in https://github.com/apple/swift/pull/71840#discussion_r1523794508.

The C stdlib headers introduce a problem: importing bionic_assert will go to the assert.h in this directory, it will not import the C++ or clang assert.h which is what users should see. I've started to use this pattern:

1. The real headers go in underscore modules like `_assert` and `_float`
2. Make cover modules that will import the C++ and clang modules.
```
module assert_h [system] {
  header "_modules/_assert_h.h"
  export *
}
```
3. Make cover headers named `_modules/_assert_h.h` to import all of the appropriate modules
```
#if !__building_module(assert_h)
#error "Do not include this header directly, include <assert.h> instead"
#endif
#include <assert.h>
```
4. Overlay the _modules directory into the SDK along with the module map.

When users `import assert_h` they will get the C++ assert.h module, the clang assert.h module, and the OS/SDK assert.h module (`_assert`)

Reply:
I see. We don't really care about specifically importing individual Libc submodules , so would it be  fine to just underscore all these C modules here? And then the user would need to import something like `Android` / `Libc` if they want to use these things from Swift directly without importing other C/C++ modules.

Reply:
That leads to what I was saying in https://github.com/apple/swift/pull/71840#discussion_r1526695720

Personally I prefer having cover modules for each of the headers because they aren't that related to each other. e.g. I might need something out of string.h or time.h but I don't want to pull in anything from signal.h or stdlib.h. There's also the performance concern: a `Libc` module will end up containing a lot of headers, e.g. on Apple platforms just stdlib.h pulls in 75 headers. That's a lot of headers to stat and open when all someone wanted was e.g. `size_t`, and it gets real expensive when NFS gets involved. POSIX is much worse, and probably shouldn't be used all that often from ObjC or Swift anyway.

That's my opinion at least. I get the convenience of having to import only one or two things and you get everything you need, but you also get a huge pile of junk in your autocomplete that you don't need, and it slows down builds too.